docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md

```diff
@@ -31,7 +31,56 @@ First, you need to create a Google OAuth Client:
 1. Click Create
 1. Copy the Client ID and Client Secret from the 'OAuth Client' modal
 
-## Enable Google OAuth in Grafana
+## Configure Google authentication client using the Grafana UI
+
+{{% admonition type="note" %}}
+Available in Public Preview in Grafana 10.4 behind the `ssoSettingsApi` feature toggle.
+{{% /admonition %}}
+
+As a Grafana Admin, you can configure generic OAuth2 client from within Grafana using the Google UI. To do this, navigate to **Administration > Authentication > Google** page and fill in the form. If you have a current configuration in the Grafana configuration file then the form will be pre-populated with those values otherwise the form will contain default values.
+
+After you have filled in the form, click **Save** to save the configuration. If the save was successful, Grafana will apply the new configurations.
+
+In case you would like to reset your changes you made on the UI back to the default values, click **Reset**. After you have reset the changes, Grafana will apply the configuration from the Grafana configuration file (if there is any configuration) or the default values.
+
+{{% admonition type="note" %}}
+If you run Grafana in high availability mode, it can happen that the configuration is not applied to all Grafana instances immediately. In this case you need to wait a minute to let the configuration propagate to all Grafana instances.
+{{% /admonition %}}
+
+Refer to [configuration options]({{< relref "#configuration-options" >}}) for more information.
+
+## Configure Google authentication client using the Terraform provider
+
+{{% admonition type="note" %}}
+Available in Public Preview in Grafana 10.4 behind the `ssoSettingsApi` feature toggle. Supported in the Terraform provider since v<TODO:TF provider version>
+{{% /admonition %}}
+
+```terraform
+resource "grafana_sso_settings" "google_sso_settings" {
+  provider_name = "google"
+  oauth2_settings {
+    name            = "Google"
+    client_id       = "CLIENT_ID"
+    client_secret   = "CLIENT_SECRET"
+    allow_sign_up   = true
+    auto_login      = false
+    scopes          = "openid email profile"
+    allowed_domains = "mycompany.com mycompany.org"
+    hosted_domain   = "mycompany.com"
+    use_pkce        = true
+  }
+}
+```
+
+Go to [Terraform Registry](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings) for a complete reference on using the `grafana_sso_settings` resource.
+
+## Configure Google authentication client using the Grafana configuration file
+
```

Comment:
Please add a sentence or 2, we do not want headings without anything underneath.

Reply:
Changed the structure a little bit (removed the `Before you begin` subtitle).